### PR TITLE
Fail2ban0.10

### PR DESCRIPTION
**Pull request by stephdl**
- Update configurations to be compatible with fail2ban-0.10
- Code refactors

https://github.com/NethServer/dev/issues/5943

### CODE REVIEW

File: nethserver-fail2ban.spec [hunk 1]
```diff
@@ -8,7 +8,8 @@ Source1: %{name}-cockpit.tar.gz
 BuildArch: noarch
 URL: http://dev.nethserver.org/projects/nethforge/wiki/%{name}
 BuildRequires: nethserver-devtools
-Requires: fail2ban fail2ban-shorewall perl-Email-Valid
+Requires: fail2ban > 0.10
```
Comment (filippocarletti) [inline nethserver-fail2ban.spec]: Shouldn't it be >= 0.10?

Comment (stephdl) [reply nethserver-fail2ban.spec]: yep, changing it, it works because I use fail2ban 0.10.4, but it could be more clear

**Comment by stephdl** [comment]
Bad new @gsanchietti we have an issue with ipset and perpetual recidive perpetual (value is `-1`), the timeout of ipset is between 0-2147483

```
2019-11-11 21:14:55,453 fail2ban.utils          [22115]: #39-Lev. 7f8732373030 -- exec: ipset add f2b-recidive 94.177.250.221 timeout -1 -exist
2019-11-11 21:14:55,453 fail2ban.utils          [22115]: ERROR   7f8732373030 -- stderr: "ipset v7.1: Syntax error: '-1' is out of range 0-2147483"
```

**Comment by stephdl** [comment]
From man ipset

timeout
All set types supports the optional timeout parameter when creating a set and adding entries. The value of the timeout parameter for the create command means the default timeout value (in seconds) for new entries. If a set is created with timeout support, then the same timeout option can be used to specify non-default timeout values when adding entries. Zero timeout value means the entry is added permanent to the set. The timeout value of already added elements can be changed by re-adding the element using the -exist option. The largest possible timeout value is 2147483 (in seconds). Example:

    ipset create test hash:ip timeout 300 
    ipset add test 192.168.0.1 timeout 60 
    ipset -exist add test 192.168.0.1 timeout 600 

When listing the set, the number of entries printed in the header might be larger than the listed number of entries for sets with the timeout extensions: the number of entries in the set is updated when elements added/deleted to the set and periodically when the garbage collector evicts the timed out entries.


I need to put 0 and test it

**Pull request merged**